EventProcessor/EventProcessor.WebJob/Processors/MessageFeedbackProcessor.cs

```diff
@@ -80,116 +73,88 @@ public void Stop()
             {
                 throw new ArgumentNullException("token");
             }
 
-            serviceClient = null;
+            ServiceClient serviceClient = null;
             try
             {
-                serviceClient =
-                    ServiceClient.CreateFromConnectionString(
-                        _iotHubConnectionString);
-
+                serviceClient = ServiceClient.CreateFromConnectionString(_iotHubConnectionString);
                 await serviceClient.OpenAsync();
 
                 while (!token.IsCancellationRequested)
                 {
-                    batchReceiver = serviceClient.GetFeedbackReceiver();
-                    batch = 
-                        await batchReceiver.ReceiveAsync(
-                            TimeSpan.FromSeconds(10.0));
+                    var batchReceiver = serviceClient.GetFeedbackReceiver();
+                    var batch = await batchReceiver.ReceiveAsync(TimeSpan.FromSeconds(10.0));
 
-                    if ((batch == null) ||
-                        ((records = batch.Records) == null))
+                    IEnumerable<FeedbackRecord> records;
+                    if ((batch == null) || ((records = batch.Records) == null))
                     {
                         continue;
                     }
 
-                    records = 
-                        records.Where(
-                            t => 
-                                (t != null) &&
-                                !string.IsNullOrEmpty(t.DeviceId) &&
-                                !string.IsNullOrEmpty(t.OriginalMessageId));
+                    records = records.Where(t => t != null)
+                        .Where(x => !string.IsNullOrEmpty(x.DeviceId))
+                        .Where(x => !string.IsNullOrEmpty(x.OriginalMessageId));
+
                     foreach (FeedbackRecord record in records)
                     {
-                        device = 
-                            await _deviceLogic.GetDeviceAsync(record.DeviceId);
-                        if (device == null)
-                        {
-                            continue;
-                        }
-
-                        Trace.TraceInformation(
+                        UpdateDeviceRecord(record, batch.EnqueuedTime);
+                    }
+
+                    await batchReceiver.CompleteAsync(batch);
+                }
+            }
+            catch (Exception ex)
+            {
+                Trace.TraceError("Error in MessageFeedbackProcessor.RunProcess, Exception: {0}", ex.Message);
+            }
+            finally
+            {
+                if (serviceClient != null)
+                {
+                    await serviceClient.CloseAsync();
+                }
+
+                _isRunning = false;
+            }
+            
+        }
+
+        private async void UpdateDeviceRecord(FeedbackRecord record, DateTime enqueuDateTime)
+        {
+            Trace.TraceInformation(
```

Comment:
Inlined the trace with UpdateDeviceRecord that performs the action instead of the foreach which basically invokes. Allows keeping the trace and action close together instead of spreading it accross different boundaries.